packages/teleport-shared/src/utils/ast-jsx-utils.ts

```diff
@@ -146,3 +147,16 @@ export const renameJSXTag = (jsxTag: types.JSXElement, newName: string, t = type
     jsxTag.closingElement.name = t.jsxIdentifier(newName)
   }
 }
+
+export const createComponentDecorator = (name: string, t = types) => {
```

Comment:
if we pass the `{ tag: ..., shadow: true}` object as a param, can't we reuse this for angular and stencil?

Reply:
Yeah, we can sent the props and the function can take care of looping and generating